### PR TITLE
Adds the living beefman plushie to the game

### PR DESCRIPTION
**Pull request by Ebb-Real**
## About The Pull Request

TG re-added the species restricted traitor items.

 This PR adds the living beefplushie to the game. They spawn in the beefstation ruins on icebox and space. Beefmen can buy this as a species-restricted traitor item too.

The living beefman plushie is a plushie that when used in hand creates a piece of meat. It has a 1 minute cooldown.

## Why It's Good For The Game

Beefstation ruins will be more unique and they now have a special item that adds more incentive for the main station to collaborate with the beefman stations.

Adds a special item for beefman traitors to have fun and possibly do gimmicks with.

## Changelog

:cl:
add: Adds the living beefman plushie, a plushie that can create meat by using in hand if given the time for its cooldown.
/:cl:

### CODE REVIEW

**Comment by StoneMason** [comment]
Put them in the beefmen spawners.

File: fulp_modules/features/toys/code/plushies.dm [hunk 1]
```diff
@@ -87,3 +87,21 @@
 	icon = 'fulp_modules/features/toys/icons/toys.dmi'
 	icon_state = "beefman"
 	squeak_override = list('sound/effects/meatslap.ogg'=1)
+
+/obj/item/toy/plush/beefplushie/living_beefplushie
+	desc = "It looks oddly alive. You feel like you should pet it."
+	COOLDOWN_DECLARE(beefplushie_cooldown)
+	var/beefplushie_cooldown_time = 2 MINUTES
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/plushies.dm:94]: This should be a define

File: fulp_modules/features/toys/code/plushies.dm [hunk 2]
```diff
@@ -87,3 +87,21 @@
...
+
+/obj/item/toy/plush/beefplushie/living_beefplushie/attack_self(mob/user)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, beefplushie_cooldown))
+		user.balloon_alert(user, "[src] isn't ready yet!")
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/plushies.dm:99]:
```suggestion
		user.balloon_alert(user, "not ready yet!")
```

File: fulp_modules/features/toys/code/plushies.dm [hunk 3]
```diff
@@ -87,3 +87,21 @@
...
+		return
+	user.balloon_alert(user, "producing meat")
+	if(!do_after(user, 3 SECONDS, target = src))
+		user.balloon_alert(user, "interrupted!")
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/plushies.dm:103]: Just remove this line we don’t need to tell them

**Comment by Ebb-Real** [comment]
what did i just do

File: fulp_modules/features/toys/code/plushies.dm (hunk 3 above)
Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm]: did the thing

File: fulp_modules/features/toys/code/plushies.dm (hunk 1 above)
Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm:94]: how do i turn it into a define

Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm:94]: new to this, just for my info. What difference does this make?

File: fulp_modules/features/toys/code/plushies.dm [hunk 4]
```diff
@@ -87,3 +87,20 @@
 	icon = 'fulp_modules/features/toys/icons/toys.dmi'
 	icon_state = "beefman"
 	squeak_override = list('sound/effects/meatslap.ogg'=1)
+
+/obj/item/toy/plush/beefplushie/living_beefplushie
+	desc = "It looks oddly alive. You feel like you should pet it."
+	COOLDOWN_DECLARE(beefplushie_cooldown)
+	#define BEEFPLUSHIE_COOLDOWN_TIME (2 MINUTES)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/plushies.dm:94]: Put the define above the code and #undef it at the end

**Comment by OneAsianTortoise** [comment]
Where is Ebb-Fake?

File: fulp_modules/features/toys/code/plushies.dm (hunk 4 above)
Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm:94]: like at the very top or just above the obj typepath?

Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm:94]: undef inside or outside the typepath?

Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm:94]: im too new at this

Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm]: we'll see if i did it right off examples

**Comment by Ebb-Real** [comment]
> Where is Ebb-Fake?

i killed him

File: fulp_modules/features/toys/code/plushies.dm [hunk 5]
```diff
@@ -87,3 +87,25 @@
 	icon = 'fulp_modules/features/toys/icons/toys.dmi'
 	icon_state = "beefman"
 	squeak_override = list('sound/effects/meatslap.ogg'=1)
+
+//Do your cooldown changes here.
+#define BEEFPLUSHIE_COOLDOWN_TIME (2 MINUTES)
+
+/obj/item/toy/plush/beefplushie/living_beefplushie
```
Comment (SgtHunk) [inline fulp_modules/features/toys/code/plushies.dm:94]: this typepath sucks, the beefplushie is already implied by it being a subtype of /obj/item/toy/plush/beefplushie

File: fulp_modules/features/toys/code/plushies.dm [hunk 6]
```diff
@@ -87,3 +87,25 @@
...
+/obj/item/toy/plush/beefplushie/living_beefplushie/attack_self(mob/user)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, beefplushie_cooldown))
+		user.balloon_alert(user, "not ready yet!")
+		return
+	user.balloon_alert(user, "producing meat")
+	if(!do_after(user, 3 SECONDS, target = src))
+		return
+	playsound(src, "sound/effects/splat.ogg", 50)
+	new /obj/item/food/meat/slab get_turf(loc)
```
Comment (SgtHunk) [inline fulp_modules/features/toys/code/plushies.dm:108]: qFZ

where's the parenthesis?

**Review by SgtHunk — CHANGES_REQUESTED** [review]
:/

File: fulp_modules/features/toys/code/plushies.dm [hunk 7]
```diff
@@ -87,3 +87,25 @@
...
+/obj/item/toy/plush/beefplushie/living/attack_self(mob/user)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, beefplushie_cooldown))
+		user.balloon_alert(user, "not ready yet!")
+		return
+	user.balloon_alert(user, "producing meat")
+	if(!do_after(user, 3 SECONDS, target = src))
+		return
+	playsound(src, "sound/effects/splat.ogg", 50)
+	new /obj/item/food/meat/slab (get_turf(loc))
```
Comment (Fikou) [inline fulp_modules/features/toys/code/plushies.dm:108]: why are you get turfing your location

**Comment by Fikou** [comment]
whats the point as an admin item it will never get use and when it does get use it has a hardcoded 2 minute cooldown so theres no purpose to use it

File: fulp_modules/features/toys/code/plushies.dm [hunk 8]
```diff
@@ -87,3 +87,25 @@
...
+	desc = "It looks oddly alive. You feel like you should pet it."
+	COOLDOWN_DECLARE(beefplushie_cooldown)
+
+//When used in hand.
+/obj/item/toy/plush/beefplushie/living/attack_self(mob/user)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, beefplushie_cooldown))
+		user.balloon_alert(user, "not ready yet!")
+		return
+	user.balloon_alert(user, "producing meat")
```
Comment (Fikou) [inline fulp_modules/features/toys/code/plushies.dm:104]: why are you making the user balloon alert themselves

**Comment by Ebb-Real** [comment]
> whats the point as an admin item it will never get use and when it does get use it has a hardcoded 2 minute cooldown so theres no purpose to use it

explained in the why its good for the game. Im not expecting massive use of this, its mostly for me to get a hang of PRs and code, thought it'd be fun to have in game. PRing it so someone else can put it in game somehow or till i learn how to or get a good idea and then if i do i actually have PR experience.

**Comment by Ebb-Real** [comment]
if the TGU gets merged i can add it as a beefman race restricted traitor item since they re-added it, cool.

**Comment by Helianthus00** [comment]
Yeah, I don't like adding things for admin only and I polled the current staff who said they likely wouldn't use it. If you add it as a race-based traitor item, I'll be more willing to consider it.

File: fulp_modules/features/toys/code/plushies.dm (hunk 8 above)
Comment (JohnFulpWillard) [reply fulp_modules/features/toys/code/plushies.dm]: this isn't resolved why did you mark it such

**Comment by Ebb-Real** [comment]
Once TGU goes through i'll start working on it.

**Comment by Ebb-Real** [comment]
or atleast attempt.

File: fulp_modules/features/toys/code/plushies.dm (hunk 8 above)
Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm]: thought i fixed it, how should i do it?

Comment (Ebb-Real) [reply fulp_modules/features/toys/code/plushies.dm]: okay cool, i think i was just blind, and forgot to fix one of them i'll fix it when i get home

**Comment by Ebb-Real** [comment]
okay i hope i did it right now

**Comment by Ebb-Real** [comment]
i'll change the title and description once i add it as a traitor item too

**Comment by Ebb-Real** [comment]
except i found out how to do drafts

**Comment by Ebb-Real** [comment]
I tried to put the useful info first and lore next in the description. Should lore disappear or is it nice?

**Review by SgtHunk — CHANGES_REQUESTED** [review]
Make the uplink item modular, and repath it. (the traitor part is implied, just beefplushie will do)

**Comment by Ebb-Real** [comment]
i dunno whats with that one failed check but i hope it dosen't matter

**Pull request merged**